code/game/objects/items/devices/flash.dm

```diff
@@ -129,9 +130,13 @@
 		if(flashfail)
 			user.visible_message("<span class='notice'>[user] fails to overload [M]'s sensors with the flash!</span>")
 		else
-			M.Knockdown(rand(5, 10))
+			length = rand(5,10)
+			M.Knockdown(length)
+			M.flashed = 1
 			M.flash_eyes(affect_silicon = 1)
 			user.visible_message("<span class='warning'>[user] overloads [M]'s sensors with the flash!</span>")
+			sleep(length*10)
```

Comment:
avoid sleeps, use spawn([amount of time]) instead then tab everything you want to happen after the amount of time. Then, check if the same still applies (in this case, check if flashed is still equal to 1)

Sleeps tend to mess with just about everything